keras_cv/layers/object_detection/anchor_generator.py

```diff
@@ -249,11 +250,14 @@ def __call__(self, image_size):
 
         stride = tf.cast(self.stride, tf.float32)
         # [W]
-        cx = tf.range(0.5 * stride, image_width + 1, stride)
+        cx = tf.range(image_width // stride)
         # [H]
-        cy = tf.range(0.5 * stride, image_height + 1, stride)
+        cy = tf.range(image_height // stride)
```

Comment:
can you explain this change a bit - perhaps a code comment?  Do we not need the +1 anymore?

Reply:
I can add a code comment. I will try to give a explanation here first, to see how to formulate it.

The general reason is, if the image `height` or `width` modulo `stride` is not equal to 0, the output shape of the anchor is not the same as from the `rpn` box predictions.

As an example of a case that worked before, say `image_height = 224` and `stride = 32` (`224%32=0`)(this was also tested out some in [this](https://github.com/keras-team/keras-cv/issues/1285#issuecomment-1400403238) issue comment):

```pycon
>>> image_height = 224
>>> stride = 32.0
>>> # old calculation
>>> tf.range(0.5 * stride, image_height + 1, stride)
<tf.Tensor: shape=(7,), dtype=float32, numpy=array([ 16.,  48.,  80., 112., 144., 176., 208.], dtype=float32)>
>>> # new calculation
>>> tf.range(image_height // stride) * stride + 0.5 * stride
<tf.Tensor: shape=(7,), dtype=float32, numpy=array([ 16.,  48.,  80., 112., 144., 176., 208.], dtype=float32)>
```
they are giving the same output, however if `stride = 64` (`224%64=32`):

```pycon
>>> image_height = 224
>>> stride = 64.0
>>> # old calculation
>>> tf.range(0.5 * stride, image_height + 1, stride)
<tf.Tensor: shape=(4,), dtype=float32, numpy=array([ 32.,  96., 160., 224.], dtype=float32)>
>>> # new calculation
>>> tf.range(image_height // stride) * stride + 0.5 * stride
<tf.Tensor: shape=(3,), dtype=float32, numpy=array([ 32.,  96., 160.], dtype=float32)>
``` 
so this results in a shape difference with `rpn`, does this explain the change for you?  I will have a think about how to formulate it in a code comment.

Also I see I should probably move `* stride + 0.5 * stride` from `meshgrid` function to `cx` and `cy` assignment.

Reply:
i see - lets move the * stride + 0.5 * stride into the range assingments for readability purposes.  Then I think LGTM.